src/gobprepare/prepare_client.py

```diff
@@ -254,6 +255,58 @@ def action_check_row_counts(self, action: RowCountConfig) -> bool:
                 )
         return tables_counts_ok
 
+    def action_check_source_sync_complete(self, action: SyncSchemaConfig) -> bool:
+        """Check source sync action. Verify when sync tables from databricks is completed.
+
+        :param action:
+        :return:
+        """
+        table_name = action["table_name"]
+        schema = action["schema"]
+
+        if not self._dst_datastore.query(check_table_existence_query("public", table_name)):  # type: ignore[union-attr]
+            raise GOBException(f"Table 'public.{table_name}' does not exists.")
+
+        self._check_last_schema_sync(table_name, schema)
+        update_columns = ["last_prepare_start = CURRENT_TIMESTAMP", "last_prepare_end = NULL"]
+        self._update_sync_schema(table_name, schema, update_columns)
+        logger.info(f"Prepare for '{schema}' started.")
+
+        return True
+
+    def action_complete_prepare(self, action: SyncSchemaConfig) -> bool:
+        """update sync table after prepare is completed.
+
+        :param action:
+        :return:
+        """
+        table_name = action["table_name"]
+        schema = action["schema"]
+        self._check_last_schema_sync(table_name, schema)
+        update_columns = ["last_prepare_end = CURRENT_TIMESTAMP"]
+        self._update_sync_schema(table_name, schema, update_columns)
+        logger.info(f"Prepare for '{schema}' completed.")
+
+        return True
+
+    def _check_last_schema_sync(self, table_name, schema) -> None:
+        select_query = f"SELECT last_sync_end, last_sync_start FROM public.{table_name} WHERE sync_schema = '{schema}'"
+        try:
+            schema_sync_data = next(self._dst_datastore.query(select_query))  # type: ignore[union-attr]
+        except StopIteration:
+            raise GOBException(f"No record for schema '{schema}' found in 'public.synced_schemas'.")
+        if schema_sync_data[0] is None:
```

Comment:
Don't know for sure, but I think schema_sync_data can be used as dict.
In that case you can use `last_sync_end` as key

Reply:
schema_sync_data is an array van alleen values.